pkg/elfwriter/writer.go

```diff
@@ -135,6 +143,58 @@ func (w *Writer) WriteProgramHeaders() {
 	}
 }
 
+// WriteSectionHeaders writes the section headers at the current location
+// and patches the file header accordingly.
+func (w *Writer) WriteSectionHeaders() {
+	shstrndx := len(w.Sections)
+	strtab := &elf.SectionHeader{
+		Name:      ".strtab",
+		Type:      elf.SHT_STRTAB,
+		Offset:    uint64(w.Here()),
+		Addralign: 1,
+	}
+	w.Sections = append(w.Sections, strtab)
+
+	strtabStart := w.Here()
+
+	strToIndex := map[string]int64{}
+
+	for _, sect := range w.Sections {
```

Comment:
Typically the first byte in the `.strtab` section is the null character, is that not necessary here?

Reply:
You are right, I missed it.